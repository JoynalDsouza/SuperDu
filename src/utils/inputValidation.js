import {isValidAlphaNumeric, isValidEmail} from './regex';

export const vaildateInput = (type, value) => {
  if (type === 'email') {
    if (!value) {
      return 'Email is required';
    }
    if (!isValidEmail(value)) {
      return 'Invalid email address';
    }
  } else if (type === 'alphaNumeric') {
    if (value?.trim()?.length === 0) {
      return 'This field is required';
    }
    if (value?.trim()?.length < 3) {
      return 'This field must be at least 3 characters';
    }
    if (!isValidAlphaNumeric(value)) {
      return 'This field must be text only';
    }
  }
  return '';
};
