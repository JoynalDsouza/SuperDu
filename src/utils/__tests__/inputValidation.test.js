import {vaildateInput} from '../inputValidation';

describe('vaildateInput', () => {
  it('take snapshot of vaildateInput', () => {
    expect(vaildateInput).toMatchSnapshot();
  });

  it('should return a error message for email', () => {
    const type = 'email';
    const value = '';
    expect(vaildateInput(type, value)).toEqual('Email is required');
  });

  it('should return a error message for invalid email', () => {
    const type = 'email';
    const value = 'test@';
    expect(vaildateInput(type, value)).toEqual('Invalid email address');
  });

  it('should return a error message for alphaNumeric', () => {
    const type = 'alphaNumeric';
    const value = '';
    expect(vaildateInput(type, value)).toEqual('This field is required');
  });

  it('should return a error message for alphaNumeric', () => {
    const type = 'alphaNumeric';
    const value = 'ab';
    expect(vaildateInput(type, value)).toEqual(
      'This field must be at least 3 characters',
    );
  });

  it('should return a error message for alphaNumeric', () => {
    const type = 'alphaNumeric';
    const value = 'ab@';
    expect(vaildateInput(type, value)).toEqual('This field must be text only');
  });
});
