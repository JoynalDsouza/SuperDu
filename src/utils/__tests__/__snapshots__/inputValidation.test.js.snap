// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vaildateInput take snapshot of vaildateInput 1`] = `[Function]`;
